src/main/java/de/tum/in/www1/artemis/repository/ProgrammingExerciseRepository.java

```diff
@@ -207,7 +223,14 @@ AND EXISTS (SELECT r.assessor FROM s.results r
      * @return the number of submissions belonging to the course id, which have the submitted flag set to true and the submission date before the exercise due date, or no exercise
      *         due date at all (only exercises with manual or semi automatic correction are considered)
      */
-    @Query("SELECT COUNT (DISTINCT p) FROM ProgrammingExerciseStudentParticipation p WHERE p.exercise.assessmentType <> 'AUTOMATIC' AND p.exercise.course.id = :#{#courseId} AND EXISTS (SELECT s FROM ProgrammingSubmission s WHERE s.participation.id = p.id AND s.submitted = TRUE)")
+    @Query("""
+            SELECT COUNT (DISTINCT p) FROM ProgrammingExerciseStudentParticipation p
+                WHERE p.exercise.assessmentType <> 'AUTOMATIC'
+                AND p.exercise.course.id = :#{#courseId}
+                AND EXISTS (SELECT s FROM ProgrammingSubmission s
```

Comment:
we should avoid sub queries due to performance issues. Can we change this query accordingly?